src/hamster/lib/desktop.py

```diff
@@ -108,13 +114,66 @@ def notify_user(self, summary="", details=""):
 
 
     def on_idle_changed(self, event, state):
+        if not self.conf_enable_timeout:
+            if '__paused__' in self._saved:
+                del self._saved['__paused__']
+            return
+        
         # state values: 0 = active, 1 = idle
-        if state == 1 and self.conf_enable_timeout:
+        if state == 1:
             idle_from = self.idle_listener.getIdleFrom()
             idle_from = timegm(idle_from.timetuple())
+            self.save_current('__paused__')
             self.storage.StopTracking(idle_from)
+        elif '__paused__' in self._saved:
+            self.restore('__paused__', delete=True)
+
+
+    def on_workspace_changed(self, event, previous_workspace, current_workspace):
+        if not self.use_workspace_changes:
+            return
+        
+        if previous_workspace is not None:
+            self.save_current( previous_workspace )
+        self.restore(current_workspace)
 
 
     def on_conf_changed(self, event, key, value):
         if hasattr(self, "conf_%s" % key):
             setattr(self, "conf_%s" % key, value)
+
+
+    def get_current(self):
+        facts = self.storage.get_todays_facts()
+        if facts:
+            last = facts[-1]
+            if not last['end_time']:
+                name = last['name']
+                category = last['category']
+                return name,category
```

Comment:
Hmm this should also include tags and other metadata, basically the whole `Fact` instance.